runtime-modules/forum/src/lib.rs

```diff
@@ -769,14 +775,34 @@ decl_module! {
             Self::ensure_is_forum_user(origin, &forum_user_id)?;
 
             // Ensure forum user is author of the thread
-            Self::ensure_is_thread_author(&thread_id, &forum_user_id)?;
+            let thread = Self::ensure_is_thread_author(&thread_id, &forum_user_id)?;
 
             // Store the event
             Self::deposit_event(RawEvent::ThreadTitleUpdated(thread_id));
 
             // Update thread title
             let title_hash = T::calculate_hash(&new_title);
-            <ThreadById<T>>::mutate(thread_id, |thread| thread.title_hash = title_hash);
+            <ThreadById<T>>::mutate(thread.category_id, thread_id, |thread| thread.title_hash = title_hash);
+
+            Ok(())
+        }
+
+        fn delete_thread(origin, moderator_id: T::ModeratorId, thread_id: T::ThreadId) -> dispatch::Result {
+            // Ensure data migration is done
+            Self::ensure_data_migration_done()?;
+
+            let thread = Self::ensure_is_thread_moderator(origin, &moderator_id, &thread_id)?;
+
+            // Delete thread
+            <ThreadById<T>>::remove(thread.category_id, thread_id);
+            <CategoryByThread<T>>::remove(thread_id);
+
+            // Delete posts
+            <PostById<T>>::iter_prefix_values(thread_id).map(|post_id| <ThreadByPost<T>>::remove(post_id));
```

Comment:
This line won't compile now. `DoubleMap` has feature [iter_prefix_values](https://crates.parity.io/frame_support/storage/trait.StorageDoubleMap.html#tymethod.iter_prefix_values) (called `iter_prefix` in the past) which is exactly what we need here, but we are using very old version of Substrate that has no such feature.

I am not sure if there is any easy way to overcome this. Is there any Substrate version upgrade planned? This problem is a blocker for this issue.

Reply:
d

Reply:
I left you a review to respond to related points, but going forward, please do not use reviews as a way of raising a point of discussion. Reviews are for one purpose only, for a maintainer to review and sign off on the quality of code. For general discussion, either just make a comment in the PR or a standalone issue, or just get in touch the relevant maintainer outside of Github.